src/check_release.rs

```diff
@@ -249,66 +250,77 @@ pub(super) fn run_check_release(
         .collect::<anyhow::Result<Vec<_>>>()?;
 
     let mut results_with_errors = vec![];
+    let mut results_with_warnings = vec![];
     for (semver_query, time_to_decide, results) in all_results {
         config
             .log_verbose(|config| {
                 let category = match overrides.effective_required_update(semver_query) {
                     RequiredSemverUpdate::Major => "major",
                     RequiredSemverUpdate::Minor => "minor",
                 };
-                if results.is_empty() {
-                    writeln!(
-                        config.stderr(),
-                        "{}{:>12}{} [{:8.3}s] {:^18} {}",
-                        Style::new()
-                            .fg_color(Some(Color::Ansi(AnsiColor::Green)))
-                            .bold(),
-                        "PASS",
-                        Reset,
-                        time_to_decide.as_secs_f32(),
-                        category,
-                        semver_query.id
-                    )?;
-                } else {
-                    writeln!(
-                        config.stderr(),
-                        "{}{:>12}{} [{:>8.3}s] {:^18} {}",
-                        Style::new()
-                            .fg_color(Some(Color::Ansi(AnsiColor::Red)))
-                            .bold(),
-                        "FAIL",
-                        Reset,
-                        time_to_decide.as_secs_f32(),
-                        category,
-                        semver_query.id
-                    )?;
-                }
+
+                let (status, status_color) = match (
+                    results.is_empty(),
+                    overrides.effective_lint_level(semver_query),
+                ) {
+                    (true, _) => ("PASS", AnsiColor::Green),
+                    (false, LintLevel::Deny) => ("FAIL", AnsiColor::Red),
+                    (false, LintLevel::Warn) => ("WARN", AnsiColor::Yellow),
+                    (false, LintLevel::Allow) => unreachable!(),
```

Comment:
Having seen and debugged a decent number of cases where an `unreachable!()` without any error message was hit, I've started to feel that it's worth spending an extra 10s to add more context to the crash (so it becomes more obvious and easily reproducible) than to spend many hours trying to reproduce a finicky error case triggered by unknown conditions.

Usually, in a situation where an unreachable statement got hit, the first change we'll make is add more context to the error so we understand what's going on. Adding that context up front doesn't meaningfully slow down development, but it makes for a much less frustrating debugging experience.
```suggestion
                    (false, LintLevel::Allow) => {
                        unreachable!(
                            "lint was unexpectedly executed at LintLevel::Allow: {semver_query:?}"
                        )
                    }
```